pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -559,5 +559,12 @@ func haveSameController(a, b metav1.Object) bool {
 		return false
 	}
 
-	return a.GetAnnotations()[RemoteControllerUID] == b.GetAnnotations()[RemoteControllerUID]
+	// NOTE(hasheddan): we set an annotation for the UID of the remote
```

Comment:
@negz note that I am still checking in `hasController` that a remote UID annotation exists, it just does not have to match. Sound ok?